x/btcstaking/types/validate_parsed_message.go

```diff
@@ -31,143 +34,302 @@ func ValidateParsedMessageAgainstTheParams(
 	stakingTxHash := pm.StakingTx.Transaction.TxHash()
 	covenantPks := parameters.MustGetCovenantPks()
 
-	// 2. Validate all data related to staking tx:
-	// - it has valid staking output
-	// - that staking time and value are correct
-	// - slashing tx is relevant to staking tx
-	// - slashing tx signature is valid
-	stakingInfo, err := btcstaking.BuildStakingInfo(
-		pm.StakerPK.PublicKey,
-		pm.FinalityProviderKeys.PublicKeys,
-		covenantPks,
-		parameters.CovenantQuorum,
-		pm.StakingTime,
-		pm.StakingValue,
-		net,
-	)
-	if err != nil {
-		return nil, ErrInvalidStakingTx.Wrapf("failed to build staking info: %v", err)
-	}
+	// handle multi-sig btc delegation and single-sig btc delegation separately
+	if pm.MultisigInfo != nil {
+		// this is the M-of-N multisig btc delegation
+		if len(pm.MultisigInfo.StakerBTCPkList.PublicKeys) < 1 || pm.MultisigInfo.StakerQuorum < 1 {
+			return nil, ErrInvalidMultisigInfo.Wrapf("number of staker btc pk list and staker quorum must be greater than 0, got: %d, %d",
+				len(pm.MultisigInfo.StakerBTCPkList.PublicKeys), pm.MultisigInfo.StakerQuorum,
+			)
+		}
 
-	stakingOutputIdx, err := bbn.GetOutputIdxInBTCTx(pm.StakingTx.Transaction, stakingInfo.StakingOutput)
+		// validate staker quorum and length of staker btc pk list doesn't exceed max M-of-N
```

Comment:
this function grown to 300 lines which  is a lot  and makes it hard to audit.


Maybe  we can move osme of this validations to smaller helper functions. For example, this one could be:
`validateStakerQuorumAgainsKeyList`  . 

Then we does't need to add comments as function names are descriptive